sanic/handlers.py

```diff
@@ -37,7 +40,18 @@ def _render_traceback_html(self, exception, request):
             uri=request.url)
 
     def add(self, exception, handler):
-        self.handlers[exception] = handler
+        self.handlers.append((exception, handler))
```

Comment:
Why do you think we need this as a dict in addition to a list? I think a dict is fine and can easily be converted to a list of tuples with, e.g. `list(self.handlers.items())`.